test/metabase/search/filter_test.clj

```diff
@@ -77,26 +79,66 @@
             [:= :card.archived false]]
            (:where (search.filter/build-filters
                     base-search-query "card"
-                    (merge default-search-ctx {:search-string "a string"}))))))
-
+                    (merge default-search-ctx {:search-string "a string"})))))))
+
+(deftest build-created-at-filter-test
+  (testing "created-at filter"
+    (mt/with-clock #t "2023-05-04T10:02:05Z[UTC]"
+      (are [created-at expected-where]
+           (= expected-where
+              (->> (search.filter/build-filters
+                    base-search-query "card"
+                    (merge default-search-ctx {:created-at created-at}))
+                   :where
+                   ;; drop the first 2 clauses [:and [:= :card.archived false]]
+                   (drop 2)))
+           ;; absolute datetime
+           "Q1-2023"                                 [[:>= [:cast :card.created_at :date] #t "2023-01-01"]
+                                                      [:< [:cast :card.created_at :date]  #t "2023-04-01"]]
+           "2016-04-18~2016-04-23"                   [[:>= [:cast :card.created_at :date] #t "2016-04-18"]
+                                                      [:< [:cast :card.created_at :date]  #t "2016-04-24"]]
+           "2016-04-18T10:30:00~2016-04-23T11:30:00" [[:>= :card.created_at #t "2016-04-18T10:30"]
+                                                      [:< :card.created_at #t "2016-04-23T11:30:01"]]
```

Comment:
This test is still wrong. The result should be

```
[[:>= :card.created_at #t "2016-04-18T10:30:00"] [:< :card.created_at #t "2016-04-24T11:30:00"]]
```

It seems the extra second is coming from applying the `inclusive-end?` setting? I think in this case `inclusive-end? false` doesn't apply. That will be a tricky one to fix, because it applies everywhere else I think.

Reply:
I think it's correct no?

For example `2016-04-18~2016-04-23` will contain results created on the `end` date, everything from `2016-04-23T00:00:00` to `2016-04-23T23:59:59`. => `[:< 2016-04-24]`

The same logic apply for second precision.
Everything created on the `end` second will still be included. Thus the extra second at the end.

Reply:
`created_at` doesn't have second precision. In tested example, the expression should evaluate to false for `2016-04-23T11:30:00.10`, but it doesn't.

Reply:
Based on our conversation with Conor, to mimic the current behaviour of the date picker in filters, the end of the interval needs to be adjusted by one minute, not one second.

The result should be:
`[[:>= :card.created_at #t "2016-04-18T10:30:00"] [:< :card.created_at #t "2016-04-24T11:31:00"]]`

Reply:
right. Updated [8c571f5](https://github.com/metabase/metabase/pull/32836/commits/8c571f5d2a149b20b34ad28687951aa2aa819e91) !